recipes/el-secretario-notmuch

```diff
@@ -0,0 +1,4 @@
+(el-secretario-notmuch
+ :fetcher git
+ :url "https://git.sr.ht/~zetagon/el-secretario"
+ :files ("el-secretario-notmuch.el"))
```

Comment:
`el-secretario-notmuch.el` with byte-compile using Emacs 27.1:
```
el-secretario-notmuch.el:37:1:Error: Cannot open load file: No such file or directory, notmuch
```
Probably this needs a `package-requires` and a `require` on it

`el-secretario-notmuch.el` with checkdoc 0.6.2:
```
<snipped the eieio-related ones>
el-secretario-notmuch.el:89: Some lines are over 80 columns wide
```
(Possibly this one can't be helped given the length of the function(s) mentioned; fyi)

`load`-check on each file:
```
Loading el-secretario-notmuch.el
  el-secretario-notmuch.el:Error: Emacs 27.1 couldn't load:
  (file-missing "Cannot open load file" "No such file or directory" "notmuch")
```

Reply:
It's not recommended to install notmuch.el from melpa if you aren't building notmuch from master. I assume most use the notmuch installation that comes with their distro so I've avoided putting notmuch in a `package-requires` (because I assume that would install `notmuch.el` from melpa.

Reply:
From my perspective (for obvious reasons) it would be better if packages laid out the dependencies they need to be automatically built and checked.  I think inattentive users will always be at risk of accidentally changing their version of notmuch, and several [packages](https://melpa.org/#/notmuch) have listed notmuch as a dependency already.  So I strongly suggest you include it, and not worry about that.

In the meantime, if I manually install notmuch and test this recipe:

`el-secretario-notmuch.el` with byte-compile using Emacs 27.1:
```
el-secretario-notmuch.el:133:1:Warning: the following functions are not known
    to be defined: el-secretario-activate-keymap,
    el-secretario--previous-source,
    el-secretario-notmuch--open-link-for-current-email,
    el-secretario--next-source
```

Also, on closer inspection, it looks like the checkdocs I snipped above are not actually false positives (missing docstrings).

Reply:
Should have fixed these issues.